metis-authentication/metis-authentication-rest/src/main/java/eu/europeana/metis/authentication/rest/AuthenticationController.java

```diff
@@ -115,14 +118,18 @@ public MetisUser loginUser(@RequestHeader("Authorization") String authorization)
       MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE})
   @ResponseStatus(HttpStatus.NO_CONTENT)
   public void updateUserPassword(@RequestHeader("Authorization") String authorization,
+      @QueryParam("oldPassword") String oldPassword,
       @QueryParam("newPassword") String newPassword)
       throws GenericMetisException {
-    String accessToken = authenticationService
-        .validateAuthorizationHeaderWithAccessToken(authorization);//Before any action, validate token
-    if (StringUtils.isEmpty(newPassword)) {
+    if (StringUtils.isBlank(oldPassword) || StringUtils.isBlank(newPassword)) {
       throw new BadContentException("newPassword not provided");
```

Comment:
This is not an appropriate error warning if the old password is blank. Could be very confusing while debugging!